### ISSUE TITLE
Changing No.# of classes

### ISSUE THREAD

**Issue by moizashfaque** [fregu856/deeplabv3#15]
Hi, 

For a custom dataset, 
- I changed the number of classes to 7  in the deeplabv3.py file (my data has 8 classes).
- Eliminated the class_wieghts file requirement in the train.py file
- Put my data & labels into the same folder structure

I get the following error in the line  `loss = loss_fn(outputs, label_imgs)`:
_IndexError: Target 78 is out of bounds._

Can someone point out where I am going wrong?
Thanks

**Comment by vikasnr** [comment]
Any luck @moizashfaque ?

**Comment by moizashfaque** [comment]
Yes @vikasnr , The pixel values of the label image was, in my case, not same as the class. i.e. class 1 had pixel values of 78, and so on. Changing pixel values of class-1 to '1', class-2 o '2' ... class-7 to '7' in the label images resolved my error.

**Comment by vikasnr** [comment]
@moizashfaque I altered the trainIDs in before preprocessing to match with the classes I wanted. Basically keep some number <255 or just 255 for all irrelevant classes.

**Comment by moizashfaque** [comment]
@vikasnr Yes, the number has to <255, and also <number of classes.
Closing the issue.
